### PR TITLE
Deploy to new server. Bot working. Digi Type is free text searchable now.

### PR DESCRIPTION
**Pull request by alessandrostagni**

### CODE REVIEW

File: ai/data/beelzex_deck.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Beelzemon X Antibody",
+    "color": "Purple",
```
Comment (WE-Kaito) [inline ai/data/beelzex_deck.json:3]: Is this up-to-date with the current models?

File: ai/data/beelzex_deck.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    "decklist": ["BT10-081","BT10-081","BT10-081","BT12-073","BT12-073","BT12-073","BT12-073","BT12-085","BT12-085","BT12-110","BT2-068","BT2-068","BT2-068","EX2-039","EX2-044","EX2-044","EX2-044","EX2-044","EX2-071","EX2-071","EX2-071","EX2-071","P-040","P-040","P-040","P-077","P-077","P-077","P-077","ST14-01","ST14-01","ST14-01","ST14-01","ST14-02","ST14-02","ST14-02","ST14-06","ST14-06","ST14-06","ST14-06","ST14-07","ST14-07","ST14-07","ST14-07","ST14-08","ST14-08","ST14-08","ST14-08","ST14-11","ST14-11","ST14-11","ST14-11","ST14-12","ST14-12"],
+    "deckStatus": "INACTIVE"
```
Comment (WE-Kaito) [inline ai/data/beelzex_deck.json:5]: see above

File: backend/src/main/java/com/github/wekaito/backend/CardService.java [hunk 3]
```diff
@@ -93,44 +94,45 @@ void fetchCards() {
...
-                    (card.digiXros().equals("-")) ? null : card.digiXros(),
-                    (card.dnaDigivolve().equals("-")) ? null : card.dnaDigivolve(),
-                    (card.securityEffect().equals("-")) ? null : card.securityEffect(),
-                    card.restrictions(),
-                    card.illustrator()));
+        List<DigivolveCondition> digivolveConditions = card.digivolveCondition().stream()
+                .map(condition -> new DigivolveCondition(
+                        condition.color(),
+                        Integer.parseInt(condition.cost()),
+                        !condition.level().equals("Tamer") ? Integer.parseInt(condition.level()) : null
```
Comment (WE-Kaito) [inline backend/src/main/java/com/github/wekaito/backend/CardService.java:103]: Is this a workaround or a proper solution?

File: frontend/src/hooks/useStore.ts [hunk 4]
```diff
@@ -145,7 +145,7 @@ export const useStore = create<State>((set, get) => ({
         if (cardType) filteredData = filteredData.filter((card) => card.cardType === cardType);
         if (stage) filteredData = filteredData.filter((card) => card.stage === stage);
         if (attribute) filteredData = filteredData.filter((card) => card.attribute === attribute);
-        if (digiType) filteredData = filteredData.filter((card) => card.digiType?.includes(digiType));
+        if (digiType) filteredData = filteredData.filter((card) => card.digiType?.join(' ').toLowerCase().includes(digiType.toLowerCase()));
```
Comment (WE-Kaito) [inline frontend/src/hooks/useStore.ts:148]: you may change it to uppercase, just for consistency? :D

**Review by WE-Kaito — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/github/wekaito/backend/CardService.java (hunk 3 above)
Comment (alessandrostagni) [reply backend/src/main/java/com/github/wekaito/backend/CardService.java:103]: The best would be to make sure that the field is numeric.
Having said that, the only value that causes the issue in digivolution cost is "Tamer".

File: ai/data/beelzex_deck.json (hunk 1 above)
Comment (alessandrostagni) [reply ai/data/beelzex_deck.json]: Great catch.
Works fine now.
Will have to delete the bot user online for this to take effect but it's very easy.
image

**Review by WE-Kaito — APPROVED** [review]

**Pull request merged**